app/src/main/java/de/adorsys/android/smsparsertest/MainActivity.java

```diff
@@ -60,16 +60,14 @@ public void onReceive(Context context, Intent intent) {
     public void onRequestPermissionsResult(int requestCode,
                                            @NonNull String permissions[],
                                            @NonNull int[] grantResults) {
-        switch (requestCode) {
-            case SmsTool.REQUEST_CODE_ASK_PERMISSIONS:
-                // If request is cancelled, the result arrays are empty.
-                if (grantResults.length <= 0
-                        || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
-                    Toast.makeText(this, getString(R.string.warning_permission_not_granted),
-                            Toast.LENGTH_LONG).show();
-                    startActivity(new Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS,
-                            Uri.parse("package:" + getApplicationContext().getPackageName())));
-                }
+        // If request is cancelled, the result arrays are empty.
+        if (requestCode == SmsTool.REQUEST_CODE_ASK_PERMISSIONS
+                && grantResults.length <= 0
```

Comment:
I think you should apply brackets here
`if (requestCode == SmsTool.REQUEST_CODE_ASK_PERMISSIONS
      && (grantResults.length <= 0 || grantResults[0] != PackageManager.PERMISSION_GRANTED))`

Reply:
Done